### PR TITLE
refactor(github): vulnerability alerts warning

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Warns if vulnerability alerts fetching is blocked.

## Context:

Many app users need to grant new permissions but aren't aware of it.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/github/index.ts [hunk 1]
```diff
@@ -1671,14 +1671,24 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
...
+    vulnerabilityAlerts = await githubApi.queryRepoField<{
       node: VulnerabilityAlert;
     }>(query, 'vulnerabilityAlerts', {
       paginate: false,
       acceptHeader: 'application/vnd.github.vixen-preview+json',
     });
+  } catch (err) {
+    logger.warn(
+      { err },
+      'Error retrieving vulnerability alerts. Please ensure permissions have been granted.'
```
Comment (HonkingGoose) [inline lib/platform/github/index.ts:1687]: Can we explain where to grant these permissions? Or link to the docs (if we have those)?

The current message lacks enough information to resolve the problem, at least for me. I would not know what to do if I got this warning... 😄 🙈 

I searched the docs for "vulnerability" and I got this documentation:
https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts
which does not mention permissions at all.

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -2229,7 +2229,15 @@ In most cases it would not be recommended, but there are some cases such as Dock
 
 ## vulnerabilityAlerts
 
-Use this object to customise PRs that are raised when vulnerability alerts are detected (GitHub-only).
+Renovate can read from GitHub's Vulnerability Alerts and customize Pull Requests accordingly.
+For this to work, you must first ensure you have enabled "Dependency graph" and "Dependabot alerts" under the "Security & analysis" section of the repository's "Settings" tab.
+
+Additionally, if you are running Renovate in app mode the you must make sure that the app has been granted the permissions to read "Vulnerability alerts".
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2235]:
```suggestion
Additionally, if you are running Renovate in app mode then you must make sure that the app has been granted the permissions to read "Vulnerability alerts".
```

Fix typo.

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -2241,7 +2249,7 @@ For example, to configure custom labels and assignees:
 }
 ```
 
-To disable vulnerability alerts completely, configure like this:
+To disable vulnerability alerts functionality completely, configure like this:
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2252]:
```suggestion
To disable the vulnerability alerts functionality completely, configure like this:
```

Add article.

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -2229,7 +2229,15 @@ In most cases it would not be recommended, but there are some cases such as Dock
 
 ## vulnerabilityAlerts
 
-Use this object to customise PRs that are raised when vulnerability alerts are detected (GitHub-only).
+Renovate can read from GitHub's Vulnerability Alerts and customize Pull Requests accordingly.
+For this to work, you must first ensure you have enabled "Dependency graph" and "Dependabot alerts" under the "Security & analysis" section of the repository's "Settings" tab.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2233]: GitHub docs has some pages that we might want to link to:

- https://docs.github.com/en/code-security/supply-chain-security/about-the-dependency-graph#enabling-the-dependency-graph
- https://docs.github.com/en/github/administering-a-repository/managing-security-and-analysis-settings-for-your-repository (Has advice on public and private repos, so we can choose which one works with Renovate bot)

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/platform/github/index.ts [hunk 5]
```diff
@@ -1671,14 +1671,28 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
       }
     }
   }`;
-  let alerts: VulnerabilityAlert[] = [];
+  let vulnerabilityAlerts: {
+    node: VulnerabilityAlert;
```
Comment (viceice) [inline lib/platform/github/index.ts:1675]: Was this a bug?

**Review by viceice — DISMISSED** [review]

File: docs/usage/configuration-options.md (hunk 4 above)
Comment (rarkins) [reply docs/usage/configuration-options.md:2233]: We can link to each of these from "Dependency graph" and "Dependabot alerts" respectively

Comment (HonkingGoose) [reply docs/usage/configuration-options.md:2233]:
```suggestion
For this to work, you must first ensure you have enabled "[Dependency graph](https://docs.github.com/en/code-security/supply-chain-security/about-the-dependency-graph#enabling-the-dependency-graph)" and "[Dependabot alerts](https://docs.github.com/en/github/administering-a-repository/managing-security-and-analysis-settings-for-your-repository)" under the "Security & analysis" section of the repository's "Settings" tab.
```

File: lib/platform/github/index.ts (hunk 5 above)
Comment (rarkins) [reply lib/platform/github/index.ts:1675]: No, it's just that the try/catch was broken into two so there was a need to have two `let` types.

**Review by viceice — DISMISSED** [review]

File: lib/platform/github/index.ts [hunk 6]
```diff
@@ -1671,14 +1671,28 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
...
       acceptHeader: 'application/vnd.github.vixen-preview+json',
     });
+  } catch (err) {
+    logger.debug({ err }, 'Error retrieving vulnerability alerts');
+    logger.warn(
+      {
+        url:
+          'https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts',
+      },
+      'Cannot access vulnerability alerts. Please ensure permissions have been granted'
```
Comment (HonkingGoose) [inline lib/platform/github/index.ts:1691]:
```suggestion
      'Cannot access vulnerability alerts. Please ensure permissions have been granted.'
```

Maybe put a full stop at the end of the second sentence?

File: docs/usage/configuration-options.md (hunk 4 above)
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:2233]: @rarkins Is the suggestion above what you wanted?

**Comment by HonkingGoose** [comment]
@rarkins Can you apply my suggestion to add the GitHub docs links?

**Review by HonkingGoose — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**